packages/vats/src/core/chain-behaviors.js

```diff
@@ -441,7 +441,7 @@ export const publishAgoricNamesToChainStorage = async ({
 };
 
 /**
- * @deprecated use reflectAgoricNamesToChainStorage
+ * @deprecated use publishAgoricNamesToChainStorage
```

Comment:
that prompted me to realize: adding `chain` under agoricNames doesn't automatically publish it to vstorage. We need to call `E(agoricNames).publishNameHubs(..., ['chain'])`.

Want to add a test that the chain info is visible from vstorage?

Reply:
that explains the trouble I was running into.

what happens when a new chain is added, do we need to call `publishNameHubs` again?  I kind of expected agoricNames would auto-update its vstorage using the `onUpdate` callback. Should it?

Reply:
confirmed: https://github.com/Agoric/agoric-sdk/blob/e193e66fc578b1c00eda5f5ab6599f1cfbdc542f/packages/vats/src/vat-agoricNames.js#L64-L67

Doesn't adding `chain` to `agoricNamesReserved` get this auto-updater? I"ll add a test

Reply:
Once we call `E(agoricNames).publishNameHubs(..., ['chain'])`, each time a chain is added, the whole list of entries in vstorage will get updated. But just doing `E(agoricNamesAdmin).provideChild('chain')` isn't enough to get the updates started for the new namehub.